submissions/api.py

```diff
@@ -244,11 +249,19 @@ def get_submission(submission_uuid, read_replica=False):
         raise SubmissionNotFoundError(
             f"No submission matching uuid {submission_uuid}"
         ) from error
-    except Exception as exc:
+    except (Exception, RuntimeWarning) as exc:
         # Something very unexpected has just happened (like DB misconfig)
+        # or an invalid json value is found in JSONField (submission.answer)
         err_msg = f"Could not get submission due to error: {exc}"
         logger.exception(err_msg)
+
+        # Switching filterwarnings back to its default behaviour
+        warnings.filterwarnings('default')
```

Comment:
You shouldn't need this here since the `finally` block should be executed whether there was an exception or not.